graphql-spqr-spring-boot-autoconfigure/src/main/java/io/leangen/graphql/spqr/spring/web/GraphQLController.java

```diff
@@ -37,7 +37,7 @@ public Object executeJsonPost(@RequestBody GraphQLRequest requestBody,
                                 R request) {
         String query = requestParams.getQuery() == null ? requestBody.getQuery() : requestParams.getQuery();
         String operationName = requestParams.getOperationName() == null ? requestBody.getOperationName() : requestParams.getOperationName();
-        Map<String, Object> variables = requestParams.getVariables() == null ? requestBody.getVariables() : requestParams.getVariables();
+        Map<String, Object> variables = requestParams.getVariables().isEmpty() ? requestBody.getVariables() : requestParams.getVariables();
```

Comment:
Since `requestParams` and `requestBody` are exclusive, wouldn't it be better to tie everything to one condition?

e.g.:

```java
        String query;
        String operationName;
        Map<String, Object> variables;
        if (requestParams.getQuery() != null) {
            query = requestParams.getQuery();
            operationName = requestParams.getOperationName();
            variables = requestParams.getVariables();
        } else {
            query = requestBody.getQuery();
            operationName = requestBody.getOperationName();
            variables = requestBody.getVariables();
        }
        if (variables == null){
            variables = Collections.emptyMap();
        }
```

Reply:
Currently the transmission of variables in the request via puery param does not work either. When I send a request (as defined in the docu https://graphql.org/learn/serving-over-http/), the following error occurs

Field error in object 'graphQLRequest' on field 'variables': rejected value [{
  "$content": "123"
}]; codes [typeMismatch.graphQLRequest.variables,typeMismatch.variables,typeMismatch.java.util.Map,typeMismatch]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [graphQLRequest.variables,variables]; arguments []; default message [variables]]; default message [Failed to convert value of type 'java.lang.String[]' to required type 'java.util.Map'; nested exception is java.lang.IllegalStateException: Cannot convert value of type 'java.lang.String[]' to required type 'java.util.Map': no matching editors or conversion strategy found]]

```java
	@Test
	public void defaultControllerTest_GET_with_variables() throws Exception {
		mockMvc.perform(
				get("/"+apiContext)
						.param("query","query echo($content: String) {echo(content: $content)}")
						.param("variables", "{\n" +
                                "  \"$content\": \"123\"\n" +
                                "}")
						)
				.andExpect(status().isOk())
				.andExpect(content().string(containsString("Hello world")));
	}
```

Reply:
It's another problem. I wrote issue.
https://github.com/leangen/graphql-spqr-spring-boot-starter/issues/58

Reply:
> I think it's good.
> but It breaks compatibility, so We should be done carefully.
> Mixing URI queries and request body parameters looks like valid HTTP, because the specification doesn't state that they are mutually exclusive in a http server application that accepts POST requests.
> How do you think, @kaqqao ?

The spec is rather vague on some finer points and seems to indicate that mixing is allowed, so I think this PR is good. Will tackle #58 separately.